rustls/src/error.rs

```diff
@@ -413,6 +413,9 @@ pub enum CertificateError {
     /// issuer.
     BadSignature,
 
+    /// A signature inside a certificate or on a handshake was made with an unsupported algorithm.
+    UnsupportedSignatureAlgorithm,
```

Comment:
Could we have the value for the signature algorithm in here? (Plus potentially the TLS version if needed to disambiguate?)

Reply:
I think the most useful additional data would be the public key and signature `AlgorithmIdentifier `s (though not that exact type, as it is `'static`), and certainly webpki's `verify_signature` function has that data at the point it produces the source errors here.

Reply:
-> https://github.com/rustls/webpki/issues/356